docs/source/exceptions.rst

```diff
@@ -0,0 +1,40 @@
+snowflake.snowpark.exceptions
+=============================
+This package contains all Snowpark client-side exceptions.
+
+.. currentmodule:: snowflake.snowpark.exceptions
+
+
+   
+.. rubric:: Exceptions
+
+.. autosummary::
+    :toctree: api/
```

Comment:
Note: this will change the URL `https://docs.snowflake.com/en/developer-guide/snowpark/reference/python/_autosummary/snowflake.snowpark.html` by replacing `_autosummary` with `api`. I think this way makes more sense, but if it's considered breaking I can revert this change.

Reply:
I prefer the new name, but I'm not very sure about the impact of URL change. I wonder if it is possible to add redirect to the original link? 🤔

Reply:
What benefit does changing this directory name from "_autosummary" to "api" give us? Is breaking all the links and messing up peoples' bookmarks worth it? I don't know how to add a redirect in this situation.

Reply:
Is it possible that we keep `_autosummary` here, but create a new page with `api`, and redirect it to `_autosummary`?

Reply:
I think redirecting is more related to how the webpages are hosted, `sphinx` will complain if it detects duplicate page indexes/paths. I'll reach out to Mary to learn about the hosting configs.
CC: @sfc-gh-aalam